libraries/core_libs/consensus/src/dag/sortition_params_manager.cpp

```diff
@@ -155,56 +107,120 @@ void SortitionParamsManager::pbftBlockPushed(const SyncBlock& block, DbStorage::
   if (config_.changing_interval == 0) {
     return;
   }
-
-  uint16_t dag_efficiency = calculateDagEfficiency(block);
-  dag_efficiencies_.push_back(dag_efficiency);
-  const auto& period = block.pbft_blk->getPeriod();
-  db_->savePbftBlockDagEfficiency(period, dag_efficiency, batch);
-  LOG(log_dg_) << period << " pbftBlockPushed, efficiency: " << dag_efficiency / 100. << "%";
-
-  if (non_empty_pbft_chain_size % config_.changing_interval == 0) {
-    const auto params_change = calculateChange(non_empty_pbft_chain_size);
-    if (params_change) {
-      db_->saveSortitionParamsChange(period, *params_change, batch);
-      params_changes_.push_back(*params_change);
+  if (ignored_efficiency_counter_ >= config_.changing_interval - config_.computation_interval) {
+    uint16_t dag_efficiency = calculateDagEfficiency(block);
+    dag_efficiencies_.push_back(dag_efficiency);
+    const auto& period = block.pbft_blk->getPeriod();
+    LOG(log_dg_) << period << " pbftBlockPushed, efficiency: " << dag_efficiency / 100. << "%";
+
+    if (non_empty_pbft_chain_size % config_.changing_interval == 0) {
+      const auto params_change = calculateChange(period);
+      params_changes_.push_back(params_change);
+      db_->saveSortitionParamsChange(period, std::move(params_change), batch);
+      cleanup();
     }
+  } else {
+    ignored_efficiency_counter_++;
+  }
+}
 
-    cleanup(period);
+int32_t efficiencyToChange(uint16_t efficiency, uint16_t goal_efficiency) {
+  uint16_t deviation = std::abs(efficiency - goal_efficiency) * 100 / goal_efficiency;
+  // If goal is 50 % return 1% change for 40%-60%, 2% change for 30%-70% and 5% change for over that
+  if (deviation < 20) {
+    return UINT16_MAX / 100;
```

Comment:
65535 / 100 = 655, why 655 is 1% change?

Reply:
655 is 1% of the total range which is 65535